### ISSUE TITLE
Supplier cannot filter paking report according to distributor

### ISSUE THREAD

**Issue by tschumilas** [openfoodfoundation/openfoodnetwork#7650]
## Description
As a supplier,  I cannot access either packing report to pack my goods that have been ordered through a hub/markeptlace.

Supplier enterprise:   Garden Party Plants -  #333
Hub enterprises I currently supply and can't access packing reports for:  Bailey's Local Foods - #80;  Fertile Ground seedlings #1791,  Fertile Ground Farm - #237

I have tried with several independent logins.  Same result

## Expected Behavior
As a supplier,  I should be able to run a packing report --  pack by customer or pack by supplier -  to prepare orders for a hub enterprise I sell through.  (The hub determines if the supplier can see the names of buyers -  but regardless -  I should be able to see the orders for my goods)

## Actual Behaviour
The hub I sell through,  is not visible for selection in the hub dropdown on either the Pack by Customer  or Pack by Supplier report.    As a supplier, I **am a**ble to run an 'Order Cycles Supplier Totals' report -  and see my orders.  However,  as a supplier I cannot run a packing report.    This is happening with 2 hubs I supply to.   In both cases I (the supplier) have given these hubs permission to 'add to order cycle'.   In both cases these hubs have set up order cycles with,  and have sold,  my products.   In both cases these hubs have allowed permission for me (supplier) to see customer names. 

Seems a pity - considering all the time we spent to add customer names to packing reports --- if the supplier can't access the reports at all.

## Steps to Reproduce

1.set up an enterprise to supply products to a hub (sells own enterprise) -  the supplying enterprise has a separate login from the hub.
2. set up a hub (sells any) enteprise with a different login
3.the supplying enterprise gives permission to the hub to sell products through an order cycle EDIT from Rachel: even giving ALL permissions to the hub does not change anything
4.the hub lists these products in an order cycle
5. logged in as the supply enterprise -  try to run  either of the packing reports EDIT from Rachel: only orders & fulfillment reports can be run

## Animated Gif/Screenshot
![hub I sell to is not visible for packing report](https://user-images.githubusercontent.com/13171778/118395514-c9709100-b618-11eb-827c-756e44d79e6f.png)

## Workaround
The workaround we are using is to ask the hub enterprise to list the supplier as a manager.  However, this means the supplier can see and access all the hub's sales (including sales of competing suppliers) - so not ideal.     The other workaround would be for the hub to download and send the packing lists to all their suppliers --- also not ideal for hubs with over 50 suppliers.

## Severity
Its not stopping sales --  its preventing suppliers from delivering their sales.    Suggestions?  is it an S2 because there is no practical workaround?    I really don't think its a good thing to add all suppliers as managers for the hubs they supply.  In my view,  it shares sales data much too broadly.

bug-s1: a critical feature is broken: checkout, payments, signup, login
bug-s2: a non-critical feature is broken, no workaround
bug-s3: a feature is broken but there is a workaround
bug-s4: it's annoying, but you can use it
bug-s5: we can live with it, only a few users impacted

https://github.com/openfoodfoundation/openfoodnetwork/wiki/Bug-severity
-->

## Your Environment

Using Chrome - latest version

* Version used:
* Browser name and version:
* Operating System and version (desktop or mobile):

**Comment by RachL** [comment]
@tschumilas I've just tested this in FR production and it's working :thinking: could you invite me to become a manager of the producer enterprise? Thanks :)

**Comment by tschumilas** [comment]
I will -  but I don't want to waste your time -  if its working elsewhere,   and no one else here has reported it -  it makes me wonder about my login.    Its the same login that is not getting confirmation emails coincidently ---  so  let me first test with completely different logins and report back.

**Comment by RachL** [comment]
@tschumilas ok, but really don't hesitate: it does not take me much time, and you may have found a critical bug!

**Comment by tschumilas** [comment]
Which login should I use Rachel?   Meanwhile can you help with something:    Should the same list of hubs appear to a supplier enterprise for  filtering all reports?    So, as a supplier -  should I have the same filtering options if I want to run an Order Cycle supplier's total report,    versus a Packing report?   I assume 'yes' - the same hubs should appear in both filtering lists.       And to show as a filtering option -   as a supplier either I have (a) given that hub permission to add to order cycle  or (b) am a co-manager of the hub I supply.

**Comment by RachL** [comment]
@tschumilas let's use a fresh one, can you invite rachel.arnould+supplier[at]protonmail.com?

**Comment by tschumilas** [comment]
I think part of my confusion with all this is that I've got multiple enterprises on multiple logins with different permissions.....  So - I'm now going to set up a 'clean' test --- and then will add you ......    I actually am starting to think that we have an issue with permissions not being removable.  But I'm getting ahead of myself.  Clean test first and I'll add you to that.

**Comment by tschumilas** [comment]
I've invited you to manage Garden Party  and Garden Party Plants  @RachL .   Garden Party Plants supplies:  Bailey's Local Foods, Fertile Ground Seedlings,  Fertile Ground,  and Ambrosia Corner Bakery.   You'll see all these hubs (and a few others I sell to off and on,  show up in the drop down list of hubs for the Order Cycle Reports.   But for me,  they do NOT show up in the drop down list of hubs for Packing reports.    I have active orders with all these hubs -  so I should see packing details.....     Curious if you see these hubs in the dropdown.

**Comment by RachL** [comment]
You're right @tschumilas actually I tested too quickly on FR production: only order & fulfillments reports are showing distributors in the dropdown. I think this is a regression and not a feature, otherwise we wouldn't have been able to work on customer names.

I'm bumping to s2 for investigation ping @lin-d-hop if you have an idea of where this could come from. I'm also updating the issue with my tests scenarios.

**Comment by Matt-Yorkley** [comment]
So it works for managers of the distributor, but not for the suppliers? I'm taking a quick look now, apart from a tiny change to the way last names are sorted in the results (last month) it looks like there haven't been any changes to the code here for roughly a year (at east in the packing report itself).

I'll do some more digging. Any other bits of info you could give that might provide some sort of clues would be much appreciated!

**Comment by Matt-Yorkley** [comment]
Going from that screenshot in the issue; is the problem partly that the hubs in question are not showing up the the dropdown for the search filters? Those two hubs should appear in the dropdown, right?

**Comment by Matt-Yorkley** [comment]
We did test it! :sweat_smile: 

I think maybe something has changed recently in a Rails upgrade and there's a scope or a query somewhere that isn't doing exactly what it's supposed to.

**Comment by Matt-Yorkley** [comment]
Wait; you can't select the distributors, but can you select the order cycle those products were sold in?

**Comment by Matt-Yorkley** [comment]
I just set this up in my environment with a user that manages a producer only, and supplies products that have been sold in an order cycle, and I see results in that report. It seems to be working. I also see the customer names are hidden.

Looking at the code that populates the list of distributors for that dropdown, I notice two things; a) it only shows distributors the current user _manages_, and b) it's always worked like that, there have not been any changes to that code in years.

Filtering by order cycle works :+1:

So I'd say it's currently usable, but we could change the way that filter works as a general reports UX improvement? It might be a bit fiddly, and it might affect the way that filter works in other reports, but I can take a look.

**Comment by Matt-Yorkley** [comment]
I might drop this to S3 and unassign myself for now.

**Comment by RachL** [comment]
@Matt-Yorkley thanks for these investigations! Alright so actually the dropdown is okay, good to know! That being said Theresa made me manager of the producer who cannot see their orders, and the packing reports, only shows data from the Garden Party and Garden Party Plants... which are hubs this producer manages.
The order & Fulfillment reports show however that there are recent orders for other hubs this producer supplies such as Bailey's Local Foods for example.

Could there be a permission problem?

**Comment by tschumilas** [comment]
I'm confused ---   I'm a supplier to several hubs.   I need to be able to pack orders for those hubs.  OFN made it so that I could see purchaser names to make that easy.   I can see the total goods I have ordered,  for either hub -  by looking at the Order Cycle Supplier Totals report (filtering by the hub I want to see).   But now I need to pack for specific customers.  How can I do that?  I cannot filter by the hubs I sell to (they don't show in the drop down).   Being a co-manager means I would see all the sales etc of the other suppliers who sell at this hub with me (not very good).    So - best option now,  is that I filter by the specific order cycle.    Problem there is -  I didn't name the order cycle,  the hub did.  I don't know the name -  and lots of order cycle names show in that dropdown.  Always things like:  Saturday May 15 .....  (not very helpful).     Is this a wishlist item then?  To make it so that a supplier can filter their packing report by hub name?

**Comment by RachL** [comment]
@tschumilas the filtering on the web would be a new feature yes (you can filter on spreadsheet), however here you don't have access to the data in the paking report, right?

**Comment by tschumilas** [comment]
Without the hub selected,   and even without the order cycle selected,    I can download a packing report by date range,  and it pulls all the supplier data,  and yes-  any hubs are captured there.   So -  I guess its not an issue of not getting access to the data (sorry if I presentd it as such).  Its that  a supplier that sells through multiple hubs,  will need to download all the data,  and then filter the report by hub.   But yes - the data are all there.       So this is really a wishlist kind of item to make accessing data easier for suppliers (esp those who sell in multiple marketplaces/hubs).     I take the point that this isn't a new thing -    I guess I never noticed this before because until recently I was co-manager with these hubs.  But now that they are very familiar with OFN,  I removed myself,  and am now just another supplier.

It also says to me that we could give hubs/marketplaces a bit of a 'tip'  to make sure their suppliers know the name of their order cycle to make it easier for them.

I'll close this.  Sorry for your time spent -  I just thought it was something new.

**Comment by RachL** [comment]
I'm really sorry @tschumilas I must have missunderstood how to test. That being said maybe this can be kept open as a papercut. Will ask on slack.

**Comment by Matt-Yorkley** [comment]
It would be a definite improvement! We should do it :+1:
